hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -584,7 +589,7 @@ def exit_with(self, reason):
         raise RunIsComplete()
 
     def generate_new_examples(self):
-        if Phase.generate not in self.settings.phases:
+        if self.interesting_examples or Phase.generate not in self.settings.phases:
```

Comment:
Might be worth a comment here about the use case this is designed to protect (reuse from the database)

Reply:
Added a comment, and a test that generation is in fact skipped when we get a failing example from the database.